### PR TITLE
DRLK Test Plan -Aliro Update

### PR DESCRIPTION
**Pull request by sumaky**
1. Added Aliro PICS
2. Update TC-DRLK-2.1

### CODE REVIEW

File: src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml [hunk 1]
```diff
@@ -1551,20 +1551,292 @@ tests:
...
           values:
               - name: "UserIndex"
                 value: 1
 
-    - label: "Clean the created credential"
-      PICS: DRLK.S.F00 && DRLK.S.F07 && DRLK.S.C26.Rsp
-      command: "ClearCredential"
-      timedInteractionTimeoutMs: 1000
+    - label: "Step 38a: TH reads AliroReaderVerificationKey attribute from DUT"
+      PICS: PICS_SKIP_SAMPLE_APP && DRLK.S.F0d && DRLK.S.A0080
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:1574]: Why is this PICS_SKIP_SAMPLE_APP?

For CI, the PICS for `DRLK.S.F0d` and `DRLK.S.A0080` are already false, so this step would be skipped anyway, no?

Similar elsewhere here: there should be no PICS_SKIP_SAMPLE_APP anywhere.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml [hunk 2]
```diff
@@ -0,0 +1,708 @@
...
+      endpoint: 0
+      cluster: "Operational Credentials"
+      attribute: "CurrentFabricIndex"
+      response:
+          saveAs: fabric-idx
+
+    - label:
+          "Step 1b: TH sends ClearUser Command to DUT with the UserIndex as
+          0xFFFE to clear all the users"
+      PICS: PICS_SKIP_SAMPLE_APP && DRLK.S.C1d.Rsp
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml:96]: Again, no PICS_SKIP_SAMPLE_APP in here.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml [hunk 3]
```diff
@@ -0,0 +1,708 @@
...
+      arguments:
+          values:
+              - name: "SigningKey"
+                value: signingkey
+              - name: "VerificationKey"
+                value: verificationkey
+              - name: "GroupIdentifier"
+                value: groupidentifier
+              - name: "GroupResolvingKey"
+                value: ""
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml:137]: This is wrong, no?  The right thing is to not have a GroupResolvingKey at all.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml [hunk 4]
```diff
@@ -0,0 +1,708 @@
...
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml:194]: Again, should not be here.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml [hunk 5]
```diff
@@ -0,0 +1,708 @@
...
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml:259]: Should not be here.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml [hunk 6]
```diff
@@ -0,0 +1,708 @@
...
+          as 1(UserStatusEnum.OccupiedEnabled), UserType as
+          0(UserTypeEnum.UnrestrictedUser), CredentialRule as
+          0(CredentialRuleEnum.Single)"
+      PICS: PICS_SKIP_SAMPLE_APP && DRLK.S.F08 && DRLK.S.C1a.Rsp
+      command: "SetUser"
+      timedInteractionTimeoutMs: 1000
+      arguments:
+          values:
+              - name: "OperationType"
+                value: 0
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml:292]:
```suggestion
                value: DataOperationTypeEnum.Add
```
and similar elsewhere for credential types, user types, credential rules, and so on.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml [hunk 7]
```diff
@@ -0,0 +1,708 @@
...
+              - name: "Status"
+                value: 0
+
+    #NEED TO CHECK WITH VALUES
+    - label: "Step 14: TH reads the LockUserChange event list from DUT"
+      PICS: PICS_SKIP_SAMPLE_APP && DRLK.S.F0d || DRLK.S.E04
+      command: "readEvent"
+      event: "LockUserChange"
+      response:
+          value:
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml:343]: This is not what the test plan says.  The test plan says there is either no event or an event like this, right?  Same for the other events.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml [hunk 8]
```diff
@@ -0,0 +1,708 @@
...
+          "Step 28b: TH reads NumberOfCredentialsSupportedPerUser attribute and
+          saves the value as numberofcredentialsupportedperuser"
+      PICS: PICS_SKIP_SAMPLE_APP && DRLK.S.A001c
+      command: "readAttribute"
+      attribute: "NumberOfCredentialsSupportedPerUser"
+      response:
+          saveAs: numberofcredentialsupportedperuser
+
+    #############
+    #STEP 28c IF LOOP
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml:613]: That part seems like it should probably a Python test, not YAML, no?

File: src/app/tests/suites/certification/ci-pics-values [hunk 9]
```diff
@@ -1521,6 +1521,15 @@ DRLK.S.A0035=0
 #ReadOnly attributes
 DRLK.S.A0028.ReadOnly=1
 DRLK.S.A002c.ReadOnly=1
+DRLK.S.A0080.ReadOnly=0
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/ci-pics-values:1524]: Don't these need to be added in the section above under "Attributes" and actually be set to "1" here because they are in fact readonly?

File: src/app/tests/suites/certification/PICS.yaml [hunk 10]
```diff
@@ -2748,6 +2748,52 @@ PICS:
           and is ReadOnly"
       id: DRLK.S.A002c.ReadOnly
 
+    - label:
+          "Does the DUT(server) support the AliroReaderVerificationKey attribute
+          and is ReadOnly"
+      id: DRLK.S.A0080.ReadOnly
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/PICS.yaml:2754]: These are not valid PICS, right?

File: src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml (hunk 1 above)
Comment (sumaky) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml]: yes removed the PICS_SKIP_SAMPLE_APP

File: src/app/tests/suites/certification/PICS.yaml (hunk 10 above)
Comment (sumaky) [reply src/app/tests/suites/certification/PICS.yaml]: removed these PICS as they are not being used

File: src/app/tests/suites/certification/ci-pics-values (hunk 9 above)
Comment (sumaky) [reply src/app/tests/suites/certification/ci-pics-values]: The PICS are now added in the attribute section

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml (hunk 8 above)
Comment (sumaky) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml]: We will be removing 2.13 to python so removed the file from this PR

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml (hunk 2 above)
Comment (sumaky) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml]: will be covered in python test case

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml (hunk 4 above)
Comment (sumaky) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml]: will be updated in python script

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml (hunk 5 above)
Comment (sumaky) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml]: will be updated in python script

File: src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml (hunk 6 above)
Comment (sumaky) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_13.yaml]: will be updated in python script

File: src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml [hunk 11]
```diff
@@ -1551,20 +1551,288 @@ tests:
...
+      command: "ClearCredential"
+      timedInteractionTimeoutMs: 1000
+      arguments:
+          values:
+              - name: "Credential"
+                value: { CredentialType: 1, CredentialIndex: 1 }
+
+    - label:
+          "Step 37: TH sends ClearUser Command to DUT with the UserIndex as 1"
+      PICS: DRLK.S.F00 && DRLK.S.F07 && DRLK.S.C1d.Rsp
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:1565]: Test plan PICS condition only has `DRLK.S.F08(USR), DRLK.S.C1d.Rsp(ClearUser)`. `DRLK.S.F00` is not part of the step.

**Review by yinyi-hu — APPROVED** [review]

File: src/app/tests/suites/certification/PICS.yaml [hunk 12]
```diff
@@ -2735,18 +2735,64 @@ PICS:
     - label: "Does the DUT(server) support the ExpiringUserTimeOut attribute?"
       id: DRLK.S.A0035
 
+    - label:
+          "Does the DUT(server) support the AliroReaderVerificationKey attribute
+          and is ReadOnly"
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/PICS.yaml:2740]: What does ReadOnly have to do with this PICS?

Same for the other ones here.

File: src/app/tests/suites/certification/PICS.yaml [hunk 13]
```diff
@@ -2735,18 +2735,64 @@ PICS:
...
+      id: DRLK.S.A0088
+
     #
     #Server ReadOnly attribute
     #
     - label:
           "Does the DUT(server) support the EnableLocalProgramming attribute and
           is ReadOnly"
-      id: DRLK.S.A0028.ReadOnly
+      id: DRLK.S.A0028
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/PICS.yaml:2790]: There's already a PICS with id DRLK.S.A0028 defined above.  Why is this here?

File: src/app/tests/suites/certification/PICS.yaml [hunk 14]
```diff
@@ -2735,18 +2735,64 @@ PICS:
...
 
     - label:
           "Does the DUT(server) support the LocalProgrammingFeatures attribute
           and is ReadOnly"
-      id: DRLK.S.A002c.ReadOnly
+      id: DRLK.S.A002c
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/PICS.yaml:2795]: Already defined above.

File: src/app/tests/suites/certification/ci-pics-values [hunk 15]
```diff
@@ -1517,11 +1517,21 @@ DRLK.S.A0031=1
...
+DRLK.S.A0083=0
+DRLK.S.A0084=0
+DRLK.S.A0085=0
+DRLK.S.A0086=0
+DRLK.S.A0087=0
+DRLK.S.A0088=0
 
 #ReadOnly attributes
 DRLK.S.A0028.ReadOnly=1
 DRLK.S.A002c.ReadOnly=1
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/ci-pics-values:1532]: These should be removed, if you're removing them from PICS.yaml, right?

File: src/app/tests/suites/certification/PICS.yaml (hunk 13 above)
Comment (cjandhyala) [reply src/app/tests/suites/certification/PICS.yaml:2790]: This came as legacy one. we should not have touched that i.e. DRLK.S.A0028.ReadOnly to DRLK.S.A0028 will update the PR.

File: src/app/tests/suites/certification/ci-pics-values (hunk 15 above)
Comment (cjandhyala) [reply src/app/tests/suites/certification/ci-pics-values:1532]: once we fix the above one, we don't need to touch this one.

File: src/app/tests/suites/certification/PICS.yaml (hunk 12 above)
Comment (cjandhyala) [reply src/app/tests/suites/certification/PICS.yaml:2740]: Nothing. its unnecessary, will fix this one.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml [hunk 16]
```diff
@@ -1551,20 +1551,288 @@ tests:
       response:
           value: NumberOfCredentialsSupportedPerUserValue
 
-    - label: "Cleanup the created user"
-      PICS: DRLK.S.F00 && DRLK.S.F07
+    - label: "Step 36: TH sends ClearCredential Command to DUT"
+      PICS: DRLK.S.F00 && DRLK.S.F07 && DRLK.S.C26.Rsp
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:1555]: Pre-existing, kind of, but this does not match the test plan.  Test plan has F08, not F07.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml [hunk 17]
```diff
@@ -1551,20 +1551,288 @@ tests:
...
+      command: "ClearCredential"
+      timedInteractionTimeoutMs: 1000
+      arguments:
+          values:
+              - name: "Credential"
+                value: { CredentialType: 1, CredentialIndex: 1 }
+
+    - label:
+          "Step 37: TH sends ClearUser Command to DUT with the UserIndex as 1"
+      PICS: DRLK.S.F07 && DRLK.S.C1d.Rsp
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:1565]: Again, pre-existing, but test plan has F08, not F07.

**Review by andy31415 — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml (hunk 17 above)
Comment (sumaky) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:1565]: There is an issue for the follow up
https://github.com/project-chip/connectedhomeip/issues/34710

File: src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml (hunk 16 above)
Comment (cjandhyala) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:1555]: will be fixed in a follow-up PR, tracking issue https://github.com/project-chip/connectedhomeip/issues/34710
